cmd/microcloud/main_init.go

```diff
@@ -533,8 +533,8 @@ func (c *initConfig) validateSystems(s *service.Handler) (err error) {
 			return fmt.Errorf("OVN underlay IP %q is invalid", sys.OVNGeneveAddr)
 		}
 
-		if subnet.Contains(underlayIP) {
-			fmt.Printf("Warning: OVN underlay IP (%s) is shared with the Ceph cluster network (%s)\n", underlayIP.String(), subnet.String())
+		if sys.OVNGeneveIface == sys.MicroCephInternalNetworkIface {
+			fmt.Printf("Warning: OVN underlay (IP: %s) is shared on the same network interface %q with the Ceph cluster network (IP: %s)\n", underlayIP.String(), sys.OVNGeneveIface, subnet.String())
```

Comment:
`subnet` above seems to be redundant now as you can just use `sys.MicroCephInternalNetworkSubnet` instead of `subnet.String()`.